tests/unit/models/transactions/test_deposit_preauth.py

```diff
@@ -2,27 +2,163 @@
 
 from xrpl.models.exceptions import XRPLModelException
 from xrpl.models.transactions import DepositPreauth
+from xrpl.models.transactions.deposit_preauth import Credential
 
 _ACCOUNT = "r9LqNeG6qHxjeUocjvVki2XR35weJ9mZgQ"
 _FEE = "0.00001"
 _SEQUENCE = 19048
 
 
 class TestDepositPreauth(TestCase):
-    def test_authorize_unauthorize_both_set(self):
-        with self.assertRaises(XRPLModelException):
+    def test_all_input_combinations(self):
+        # this value is used for authorized_credentials or unauthorized_credentials
+        # fields
+        sample_credentials = [
+            Credential(issuer="SampleIssuer", credential_type="SampleCredType")
+        ]
+        for val in range(0, 16):
```

Comment:
I get what you're trying to accomplish here, but this test is really hard to read and IMO a bit unnecessary.

Reply:
Hmm, I agree. Shall I test two-three cases where the inputs will throw an error? Will that suffice?

Reply:
Yes. The tests don't need to be fully exhaustive, as long as you're getting the code coverage.

Reply:
okay, simplified the tests in 379620a5